sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -63,7 +64,7 @@ class SparkSession private(
     @transient private val existingSharedState: Option[SharedState])
   extends Serializable with Logging { self =>
 
-  def this(sc: SparkContext) {
+  private[sql] def this(sc: SparkContext) {
```

Comment:
yeah I think the problem is that in `getOrCreate` right now we go the other way, i.e. getting the session from the context